pkg/provider/azure_controller_common.go

```diff
@@ -254,8 +262,12 @@ func (c *controllerCommon) AttachDisk(ctx context.Context, async bool, diskName,
 	c.diskStateMap.Store(disk, "attaching")
 	defer c.diskStateMap.Delete(disk)
 	future, err := vmset.AttachDisk(ctx, nodeName, diskMap)
-	if err != nil {
-		return -1, err
+	if future == nil {
+		return -1, fmt.Errorf("nil future was returned: %v", err)
+	} else if err != nil {
+		if !configAccepted(future) {
```

Comment:
if there is 2XX return code with `err != nil`, but it's not `OperationPreemptedErrorCode`, in current PR, it would still call `vmset.WaitForUpdateResult`, is that expected?

Reply:
Given the implementation of polling tracker and how async operations are polled, we expect the `WaitForUpdateResult` to fail with the same error, which we intend to handle in the retry loop.
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/5e5c4af9bee955f0a266f6b8182ca0221044cd1d/pkg/provider/azure_controller_common.go#L290-L301

But I think we could add an additional check to be safe and only `WaitForUpdateResult` after `vmset.AttachDisk` if the operation did not result in error.